apolo_kube_client/_core.py

```diff
@@ -0,0 +1,229 @@
+        self._cert_authority_path = config.cert_authority_path
+
+        if config.auth_type == KubeClientAuthType.TOKEN:
+            assert config.token or config.token_path
+        elif config.auth_type == KubeClientAuthType.CERTIFICATE:
+            assert config.auth_cert_path
+            assert config.auth_cert_key_path
+
+        self._auth_type = config.auth_type
+        self._auth_cert_path = config.auth_cert_path
+        self._auth_cert_key_path = config.auth_cert_key_path
+        self._token = config.token
+        self._token_path = config.token_path
+        self._token_update_interval_s = config.token_update_interval_s
+
+        self._conn_timeout_s = config.client_conn_timeout_s
+        self._read_timeout_s = config.client_read_timeout_s
+        self._watch_timeout_s = config.client_watch_timeout_s
+        self._conn_pool_size = config.client_conn_pool_size
+        self._trace_configs = trace_configs
+
+        self._client: aiohttp.ClientSession | None = None
+        self._token_updater_task: asyncio.Task[None] | None = None
+
+    def __str__(self) -> str:
+        return self.__class__.__name__
+
+    async def __aenter__(self) -> Self:
+        await self.init()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: BaseException | None,
+        exc_tb: TracebackType | None,
+    ) -> None:
+        await self.close()
+
+    async def init(self) -> None:
+        logger.info("%s: initializing", self)
+        if self._token_path:
+            self._refresh_token_from_file()
+            self._token_updater_task = asyncio.create_task(self._start_token_updater())
+
+        connector = aiohttp.TCPConnector(
+            limit=self._conn_pool_size, ssl=self._create_ssl_context()
+        )
+        timeout = aiohttp.ClientTimeout(
+            connect=self._conn_timeout_s, total=self._read_timeout_s
+        )
+        self._client = aiohttp.ClientSession(
+            connector=connector,
+            timeout=timeout,
+            trace_configs=self._trace_configs,
+            raise_for_status=self._raise_for_status,
+        )
+
+    async def close(self) -> None:
+        logger.info("%s: closing", self)
+        if self._client:
+            await self._client.close()
+            self._client = None
+        if self._token_updater_task:
+            self._token_updater_task.cancel()
+            with suppress(asyncio.CancelledError):
+                await self._token_updater_task
+            self._token_updater_task = None
+        logger.info("%s: closed", self)
+
+    @property
+    def base_url(self) -> URL:
+        return self._base_url
+
+    @property
+    def namespace(self) -> str:
+        return self._namespace
+
+    async def request(
+        self,
+        method: str,
+        url: URL | str,
+        params: dict[str, Any] | None = None,
+        json: dict[str, Any] | None = None,
+        **kwargs: Any,
```

Comment:
`kwargs` still exists.
And `Any` used many times. Let's discuss it after the next daily.